### PR TITLE
Discord Sample Bundle

### PR DESCRIPTION
**Pull request by Tim-Tech-Dev**
- README Update
- Discord sample bundle
- Discord sample bundle Documentation
- Uniform sample bundle naming

### CODE REVIEW

**Comment by Tim-Tech-Dev** [comment]
Lerna does not link the local dependencies discord & core to the sample -> tsc crashes
But only with a fresh install.

**Comment by Tim-Tech-Dev** [comment]
```
lerna info Bootstrapping 7 packages
lerna info Installing external dependencies
lerna ERR! npm install exited 1 in 'discord-guild-chat'
lerna ERR! npm install stderr:
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/nodecg-io-core - Not found
npm ERR! 404 
npm ERR! 404  'nodecg-io-core@0.1.0' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
npm ERR! 404 
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.
```

**Comment by daniel0611** [comment]
@Tim-Tech-Dev pinning the version in the `package.json` and deleting the `package-lock.json` to regenerate it may fix this.

**Comment by daniel0611** [comment]
I've just tested it locally and regenerating the package-lock.json does fix the issue. 
I've always had this when I installed a package using `npm i some-package` in some package of this lerna project.
Note (to probably all): Running `npm i some-package` in a package of a lerna project creates problems, use [lerna add](https://github.com/lerna/lerna/tree/master/commands/add) or add the dependency manually to the package.json file and use `lerna bootstrap` or the postinstall script in the root of the repo. As an example I saw @derNiklaas doing this on stream with the same error message.
Still learning how to use lerna....

**Comment by Tim-Tech-Dev** [comment]
Thank you for your help!

File: docs/services/twitch.md [hunk 1]
```diff
@@ -1,8 +1,17 @@
 ## Using the Twitch sample bundle
 
-The Twitch-Chat example bundle in `samples/twitch-chat` demonstrates the ability to get access to a twitch chat and printing all messages of it. Here is a guide to how to get it working:
+The Twitch-chat example bundle in `samples/twitch-chat` demonstrates the ability to get access to a twitch chat and printing all messages of it. Here is a guide to how to get it working.
 
-1. Start nodecg with nodecg-io installed. The bundle is currently part of it so it should also be loaded.
+### Prerequisites
+
+* Working NodeCG & nodcg-oi installation
```
Comment (daniel0611) [inline docs/services/twitch.md:7]:
```suggestion
* Working NodeCG & nodcg-io installation
```

File: docs/services/twitch.md [hunk 2]
```diff
@@ -1,8 +1,17 @@
...
+* a Twitch oAuth-Key
+
+*Note:* If you don't have such a Key yet you can generate it on  https://twitchapps.com/tmi/. Just log into your Twitch account and copy the token.
```
Comment (daniel0611) [inline docs/services/twitch.md:10]:
```suggestion
*Note:* If you don't have such a key yet you can generate it on https://twitchapps.com/tmi/. Just log into your Twitch account and copy the token.  
```

File: samples/discord-guild-chat/extension/index.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+
+module.exports = function (nodecg: NodeCG) {
+    nodecg.log.info("Sample bundle for discord started");
+
+    // This implicit cast determines the client type in the requireService call
+    const discord: ServiceProvider<DiscordServiceClient> | undefined = nodecg.extensions["nodecg-io-discord"] as any;
+
+
+    discord?.requireService("Discord-guild-chat", (client) => {
```
Comment (daniel0611) [inline samples/discord-guild-chat/extension/index.ts:13]: Developer UX problem: these bundle names should be the same as the ones in the `package.json` of the corresponding bundle. Maybe we should require the nodecg instance instead, which contains a variable with the bundle name. That way developers can't mess this up.
```suggestion
    discord?.requireService("discord-guild-chat", (client) => {
```

File: samples/twitch-chat/extension/index.ts [hunk 4]
```diff
@@ -13,7 +13,7 @@ module.exports = function (nodecg: NodeCG) {
     // Note that this does need a # before the channel name and is case-insensitive.
     const twitchChannels = ["#skate702", "#daniel0611"];
 
-    twitch?.requireService("sample", (client) => {
+    twitch?.requireService("Twitch-chat", (client) => {
```
Comment (daniel0611) [inline samples/twitch-chat/extension/index.ts:16]: Same as with discord.
```suggestion
    twitch?.requireService("twitch-chat", (client) => {
```

**Comment by Tim-Tech-Dev** [comment]
Should now be complete and up to date

**Pull request closed without merging**